pkg/addons/default/aws_node_test.go

```diff
@@ -91,97 +98,97 @@ var _ = Describe("default addons - aws-node", func() {
 		})
 
-		It("can update 1.15 sample for china region to multi-architecture image", func() {
-			rawClient.ClientSetUseUpdatedObjects = false // must be set for subsequent UpdateAWSNode
+		When("using a chinese region", func() {
+			It("updates it and uses the amazonaws.com.cn address", func() {
+				input.Region = "cn-northwest-1"
 
-			preUpdateAwsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), AWSNode, metav1.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
-			_, err = UpdateAWSNode(rawClient, "cn-northwest-1", false)
-			Expect(err).NotTo(HaveOccurred())
-
-			rawClient.ClientSetUseUpdatedObjects = true // for verification of updated objects
+				_, err := da.UpdateAWSNode(input, false)
+				Expect(err).NotTo(HaveOccurred())
 
-			awsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), AWSNode, metav1.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
-			Expect(awsNode.Spec.Template.Spec.Containers).To(HaveLen(1))
-			Expect(awsNode.Spec.Template.Spec.Containers[0].Image).NotTo(
-				Equal(preUpdateAwsNode.Spec.Template.Spec.Containers[0].Image),
-			)
-			Expect(awsNode.Spec.Template.Spec.InitContainers).To(HaveLen(1))
-			Expect(awsNode.Spec.Template.Spec.InitContainers[0].Image).To(
-				HavePrefix("961992271922.dkr.ecr.cn-northwest-1.amazonaws.com.cn/amazon-k8s-cni-init"),
-			)
+				awsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), da.AWSNode, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(awsNode.Spec.Template.Spec.Containers).To(HaveLen(1))
+				Expect(awsNode.Spec.Template.Spec.Containers[0].Image).To(
+					Equal("961992271922.dkr.ecr.cn-northwest-1.amazonaws.com.cn/amazon-k8s-cni:v1.9.3"),
+				)
+				Expect(awsNode.Spec.Template.Spec.InitContainers).To(HaveLen(1))
+				Expect(awsNode.Spec.Template.Spec.InitContainers[0].Image).To(
+					Equal("961992271922.dkr.ecr.cn-northwest-1.amazonaws.com.cn/amazon-k8s-cni-init:v1.9.3"),
+				)
+			})
 		})
 
-		It("detects matching image version when determining plan", func() {
-			// updating from latest to latest needs no updating
-			needsUpdate, err := UpdateAWSNode(rawClient, "eu-west-2", true)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(needsUpdate).To(BeFalse())
+		When("dry run is true", func() {
+			When("it needs an update", func() {
+				It("returns true", func() {
+					needsUpdate, err := da.UpdateAWSNode(input, true)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(needsUpdate).To(BeTrue())
+
+					awsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), da.AWSNode, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					//should be unchanged
+					Expect(awsNode.Spec).To(Equal(preUpdateAwsNode.Spec))
+				})
+			})
+
+			When("it doesn't need an update", func() {
+				BeforeEach(func() {
+					rawClient = testutils.NewFakeRawClient()
```

Comment:
This before is repeated over and over again in other tests as well. Might be worth putting it into a function and calling it with different params?